hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/meta/TransactionMetadataBuilder.java

```diff
@@ -202,6 +245,19 @@ private boolean designatesAccountRemoval(@NonNull final AccountID id) {
                 && id.getAlias().isEmpty();
     }
 
+    /**
+     * Checks if the contractId is a sentinel id <code>0.0.0</code>
+     *
+     * @param id given contractId
+     * @return true if the given contractId is
+     */
+    private boolean designatesContractRemoval(@NonNull final ContractID id) {
+        return id.getShardNum() == 0
```

Comment:
Same here: Since Id is accessed I would do a null check

Reply:
Same here